ibis/backends/dask/execution/selection.py

```diff
@@ -137,10 +136,16 @@ def build_df_from_projection(
     # Slow path when we cannot do direct assigns
     # Create a unique row identifier and set it as the index. This is
     # used in dd.concat to merge the pieces back together.
-    data = add_partitioned_sorted_column(data)
-    data_pieces = [compute_projection(node, op, data, **kwargs) for node in selections]
+    partitioned_data = add_partitioned_sorted_column(data)
+
+    data_pieces = [
+        compute_projection(node, op, partitioned_data, **kwargs) for node in selections
+    ]
 
-    return dd.concat(data_pieces, axis=1).reset_index(drop=True)
+    result = dd.concat(data_pieces, axis=1)
+    result.index = result.index.rename(None)
```

Comment:
Why do we need to drop the index name?

Reply:
The point of calling `reset_index(drop=True)` here is to drop `_ibis_index` and give it an arbitrary new index. The dataframe here is already nicely partitioned on `_ibis_index`.
It would be a waste to drop it, so instead, we just drop the name and use that as the dataframe index.